### PR TITLE
shipping methods plugin implementation

### PR DESCRIPTION
**Pull request by rajeev-k-tomy**

### CODE REVIEW

File: src/reactapp/src/components/CheckoutForm/CheckoutFormWrapper.jsx [hunk 1]
```diff
@@ -36,50 +33,59 @@ function CheckoutFormWrapper({ initialData, children }) {
     }
 
     const { cart, customer } = initialData;
-    const email = _get(cart, 'email', '');
+    const email = _get(cart, 'email') || '';
     const appliedCoupon = _get(cart, 'appliedCoupon');
```
Comment (wigman) [inline src/reactapp/src/components/CheckoutForm/CheckoutFormWrapper.jsx:37]: @rajeev-k-tomy for my understanding, coupon is the only value here that is not required, and that's why it's the only one that does not have a fallback value?

**Review by wigman — APPROVED** [review]
Looks good! Thanks a lot Rajeev!

File: src/reactapp/src/components/shippingMethod/components/ShippingMethodList.jsx [hunk 2]
```diff
@@ -44,6 +45,18 @@ function ShippingMethodList() {
         {_objToArray(methodList).map((method) => {
           const { id: methodId, carrierTitle, methodTitle, price } = method;
           const methodName = `${carrierTitle} (${methodTitle}): `;
+          const MethodRenderer = methodRenderers[methodId];
+
+          if (MethodRenderer) {
+            return (
+              <MethodRenderer
```
Comment (ArjenMiedema) [inline src/reactapp/src/components/shippingMethod/components/ShippingMethodList.jsx:52]: So you want the `<li>` element to be defined in the method renderer and not here?

File: src/reactapp/src/components/CheckoutForm/CheckoutFormWrapper.jsx (hunk 1 above)
Comment (rajeev-k-tomy) [reply src/reactapp/src/components/CheckoutForm/CheckoutFormWrapper.jsx:37]: Your assumption is right @wigman

File: src/reactapp/src/components/shippingMethod/components/ShippingMethodList.jsx (hunk 2 above)
Comment (rajeev-k-tomy) [reply src/reactapp/src/components/shippingMethod/components/ShippingMethodList.jsx:52]: I think you have a valid point. It is better we keep <li> here in this component.

**Pull request merged**